README.adoc

```diff
@@ -65,3 +65,120 @@ if (infra.isTrusted()) {
+                    'PATH+JAVA=${JAVA_HOME}/bin',
+                    "PATH+MAVEN=${tool 'mvn'}/bin"]
+          withEnv(mavenEnv) {
+            sh "mvn clean install -DskipTests"
+          }
+          dir("target") {
+           stash includes: '*.hpi', name: 'localPlugins'
+          }
+
+          runATH(metadataFile:"metadata.yml", athRevision: "master")
+        }
+    }
+----
+
+.Jenkinsfile
+[source,groovy]
+----
+    node("linux") {
+        dir("sources") {
+          checkout scm
+          List<String> mavenEnv = [
+                    "JAVA_HOME=${tool 'jdk8'}",
+                    'PATH+JAVA=${JAVA_HOME}/bin',
+                    "PATH+MAVEN=${tool 'mvn'}/bin"]
+          withEnv(mavenEnv) {
+            sh "mvn clean install -DskipTests"
+          }
+          dir("target") {
+           stash includes: '*.hpi', name: 'snapshots'
+          }
+
+          env.RUN_ATH_LOCAL_PLUGINS_STASH_NAME="snapshots"
+          runATH(metadataFile:"metadata.yml", athRevision: "master")
+        }
+    }
+----
+
+The metadata file is a YAML file with the following structure:
+
+.metadata
+[source,yaml]
+----
+ath:
+  athUrl: https://github.com/jenkinsci/acceptance-test-harness.git
+  athRevision: acceptance-test-harness-1.59
+  jenkins: 2.89.4
+  failFast: false
+  rerunFailingTestsCount: 0
+  useLocalSnapshots: true
+  browsers:
+    - firefox
+    - chrome
+  tests:
+    - Test1
+    - Test2
+    - Test3
+  categories:
+    - Category1
+    - Category2
+----
+
+Where:
+
+
+    * `athUrl`: (Optional) The URL to get the ATH sources. It can point to a local path or a github destination. If specified it will override the parameter in the runATH step
+    * `athRevision`: (Optional) The ATH revision to use can be a branch or tag name or a commit id. If specified it will override the parameter in the runATH step
+    * `jenkins`: (Optional) URI to the jenkins.war file, Jenkins version or one of "latest", "latest-rc", "lts" and "lts-rc". If specified it will override the parameter in the runATH step
+    * `failFast`: If the run has to fail fast or not. Defaults to false if not specified
+    * `rerunFailingTestsCount`: The number of runs per failing test (a la maven). Defaults to zero</li>
+    * `useLocalSnapshots`: If the ATH should use local versions of the plugins. Defaults to true. *Note that if true the runATH expects the local plugins to be stashed, setting this to true without the stash will make the step fail*
+    * `browsers`: The list of browsers to use when running ATH Defaults to firefox. *Note that currently only firefox browser is supported, any other will be ignored*
+    * `tests`: The list of tests to run for the component that calls the step. If no particular set of tests or categories is defined the SmokeTest Category of the ATH will be run
+    * `categories`: The list of Categories to run. Defaults to nothing
+
+In case you want to use the defaults for all properties you can use
```

Comment:
I would also note that an error will occur if the metadata file doesn't contain an `ath` section at all - a blank metadata file != `ath: "default"`